From 84b42215f5c707f978b8c443626e88d95bfc77f1 Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Thu, 9 Apr 2020 14:25:00 +0700
Subject: [PATCH 289/443] arm64: dts: renesas: r8a774e1: add VIN and CSI-2
 nodes

This commit adds VIN{0,1,2,3,4,5,6,7} and CSI{40,20} device nodes
support for r8a774e1 SoC with remote link:
- CSI20: link to VIN{0,1,2,3,4,5,6,7}.
- CSI40: link to VIN{0,1,2,3).

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 arch/arm64/boot/dts/renesas/r8a774e1.dtsi | 129 ++++++++++++++++++++++--------
 1 file changed, 95 insertions(+), 34 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a774e1.dtsi b/arch/arm64/boot/dts/renesas/r8a774e1.dtsi
index 9a5538a..7a0f05b 100644
--- a/arch/arm64/boot/dts/renesas/r8a774e1.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a774e1.dtsi
@@ -2292,7 +2292,14 @@
 		};
 
 		vin0: video@e6ef0000 {
+			compatible = "renesas,vin-r8a774e1";
 			reg = <0 0xe6ef0000 0 0x1000>;
+			interrupts = <GIC_SPI 188 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 811>;
+			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
+			resets = <&cpg 811>;
+			renesas,id = <0>;
+			status = "disabled";
 
 			ports {
 				#address-cells = <1>;
@@ -2306,17 +2313,27 @@
 
 					vin0csi20: endpoint@0 {
 						reg = <0>;
+						remote-endpoint= <&csi20vin0>;
 					};
 					vin0csi40: endpoint@2 {
 						reg = <2>;
+						remote-endpoint= <&csi40vin0>;
 					};
 				};
 			};
-			/* placeholder */
 		};
 
 		vin1: video@e6ef1000 {
+			compatible = "renesas,vin-r8a774e1";
 			reg = <0 0xe6ef1000 0 0x1000>;
+			interrupts = <GIC_SPI 189 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 810>;
+			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
+			resets = <&cpg 810>;
+			renesas,id = <1>;
+			status = "disabled";
+
+			ports {
 				#address-cells = <1>;
 				#size-cells = <0>;
 
@@ -2328,16 +2345,25 @@
 
 					vin1csi20: endpoint@0 {
 						reg = <0>;
+						remote-endpoint= <&csi20vin1>;
 					};
 					vin1csi40: endpoint@2 {
 						reg = <2>;
+						remote-endpoint= <&csi40vin1>;
 					};
 				};
-			/* placeholder */
+			};
 		};
 
 		vin2: video@e6ef2000 {
+			compatible = "renesas,vin-r8a774e1";
 			reg = <0 0xe6ef2000 0 0x1000>;
+			interrupts = <GIC_SPI 190 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 809>;
+			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
+			resets = <&cpg 809>;
+			renesas,id = <2>;
+			status = "disabled";
 
 			ports {
 				#address-cells = <1>;
@@ -2351,17 +2377,25 @@
 
 					vin2csi20: endpoint@0 {
 						reg = <0>;
+						remote-endpoint= <&csi20vin2>;
 					};
 					vin2csi40: endpoint@2 {
 						reg = <2>;
+						remote-endpoint= <&csi40vin2>;
 					};
 				};
 			};
-			/* placeholder */
 		};
 
 		vin3: video@e6ef3000 {
+			compatible = "renesas,vin-r8a774e1";
 			reg = <0 0xe6ef3000 0 0x1000>;
+			interrupts = <GIC_SPI 191 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 808>;
+			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
+			resets = <&cpg 808>;
+			renesas,id = <3>;
+			status = "disabled";
 
 			ports {
 				#address-cells = <1>;
@@ -2375,17 +2409,25 @@
 
 					vin3csi20: endpoint@0 {
 						reg = <0>;
+						remote-endpoint= <&csi20vin3>;
 					};
 					vin3csi40: endpoint@2 {
 						reg = <2>;
+						remote-endpoint= <&csi40vin3>;
 					};
 				};
 			};
-			/* placeholder */
 		};
 
 		vin4: video@e6ef4000 {
+			compatible = "renesas,vin-r8a774e1";
 			reg = <0 0xe6ef4000 0 0x1000>;
+			interrupts = <GIC_SPI 174 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 807>;
+			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
+			resets = <&cpg 807>;
+			renesas,id = <4>;
+			status = "disabled";
 
 			ports {
 				#address-cells = <1>;
@@ -2399,17 +2441,21 @@
 
 					vin4csi20: endpoint@0 {
 						reg = <0>;
-					};
-					vin4csi40: endpoint@2 {
-						reg = <2>;
+						remote-endpoint= <&csi20vin4>;
 					};
 				};
 			};
-			/* placeholder */
 		};
 
 		vin5: video@e6ef5000 {
+			compatible = "renesas,vin-r8a774e1";
 			reg = <0 0xe6ef5000 0 0x1000>;
+			interrupts = <GIC_SPI 175 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 806>;
+			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
+			resets = <&cpg 806>;
+			renesas,id = <5>;
+			status = "disabled";
 
 			ports {
 				#address-cells = <1>;
@@ -2423,17 +2469,21 @@
 
 					vin5csi20: endpoint@0 {
 						reg = <0>;
-					};
-					vin5csi40: endpoint@2 {
-						reg = <2>;
+						remote-endpoint= <&csi20vin5>;
 					};
 				};
 			};
-			/* placeholder */
 		};
 
 		vin6: video@e6ef6000 {
+			compatible = "renesas,vin-r8a774e1";
 			reg = <0 0xe6ef6000 0 0x1000>;
+			interrupts = <GIC_SPI 176 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 805>;
+			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
+			resets = <&cpg 805>;
+			renesas,id = <6>;
+			status = "disabled";
 
 			ports {
 				#address-cells = <1>;
@@ -2447,17 +2497,21 @@
 
 					vin6csi20: endpoint@0 {
 						reg = <0>;
-					};
-					vin6csi40: endpoint@2 {
-						reg = <2>;
+						remote-endpoint= <&csi20vin6>;
 					};
 				};
 			};
-			/* placeholder */
 		};
 
 		vin7: video@e6ef7000 {
+			compatible = "renesas,vin-r8a774e1";
 			reg = <0 0xe6ef7000 0 0x1000>;
+			interrupts = <GIC_SPI 171 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 804>;
+			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
+			resets = <&cpg 804>;
+			renesas,id = <7>;
+			status = "disabled";
 
 			ports {
 				#address-cells = <1>;
@@ -2471,17 +2525,20 @@
 
 					vin7csi20: endpoint@0 {
 						reg = <0>;
-					};
-					vin7csi40: endpoint@2 {
-						reg = <2>;
+						remote-endpoint= <&csi20vin7>;
 					};
 				};
 			};
-			/* placeholder */
 		};
 
 		csi20: csi2@fea80000 {
+			compatible = "renesas,r8a774e1-csi2";
 			reg = <0 0xfea80000 0 0x10000>;
+			interrupts = <GIC_SPI 184 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 714>;
+			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
+			resets = <&cpg 714>;
+			status = "disabled";
 
 			ports {
 				#address-cells = <1>;
@@ -2495,35 +2552,48 @@
 
 					csi20vin0: endpoint@0 {
 						reg = <0>;
+						remote-endpoint = <&vin0csi20>;
 					};
 					csi20vin1: endpoint@1 {
 						reg = <1>;
+						remote-endpoint = <&vin1csi20>;
 					};
 					csi20vin2: endpoint@2 {
 						reg = <2>;
+						remote-endpoint = <&vin2csi20>;
 					};
 					csi20vin3: endpoint@3 {
 						reg = <3>;
+						remote-endpoint = <&vin3csi20>;
 					};
 					csi20vin4: endpoint@4 {
 						reg = <4>;
+						remote-endpoint = <&vin4csi20>;
 					};
 					csi20vin5: endpoint@5 {
 						reg = <5>;
+						remote-endpoint = <&vin5csi20>;
 					};
 					csi20vin6: endpoint@6 {
 						reg = <6>;
+						remote-endpoint = <&vin6csi20>;
 					};
 					csi20vin7: endpoint@7 {
 						reg = <7>;
+						remote-endpoint = <&vin7csi20>;
 					};
 				};
 			};
-			/* placeholder */
 		};
 
 		csi40: csi2@feaa0000 {
+			compatible = "renesas,r8a774e1-csi2";
 			reg = <0 0xfeaa0000 0 0x10000>;
+			interrupts = <GIC_SPI 246 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 716>;
+			power-domains = <&sysc R8A774E1_PD_ALWAYS_ON>;
+			resets = <&cpg 716>;
+			status = "disabled";
 
 			ports {
 				#address-cells = <1>;
@@ -2537,31 +2607,22 @@
 
 					csi40vin0: endpoint@0 {
 						reg = <0>;
+						remote-endpoint = <&vin0csi40>;
 					};
 					csi40vin1: endpoint@1 {
 						reg = <1>;
+						remote-endpoint = <&vin1csi40>;
 					};
 					csi40vin2: endpoint@2 {
 						reg = <2>;
+						remote-endpoint = <&vin2csi40>;
 					};
 					csi40vin3: endpoint@3 {
 						reg = <3>;
-					};
-					csi40vin4: endpoint@4 {
-						reg = <4>;
-					};
-					csi40vin5: endpoint@5 {
-						reg = <5>;
-					};
-					csi40vin6: endpoint@6 {
-						reg = <6>;
-					};
-					csi40vin7: endpoint@7 {
-						reg = <7>;
+						remote-endpoint = <&vin3csi40>;
 					};
 				};
 			};
-			/* placeholder */
 		};
 	};
 
-- 
2.7.4

